easybuild/easyblocks/p/python.py

```diff
@@ -128,6 +129,21 @@ def __init__(self, *args, **kwargs):
             easybuild_subdir = log_path()
             self.pythonpath = os.path.join(easybuild_subdir, 'python')
 
+        ext_defaults = {
+            'download_dep_fail': True,
+            # The only installed packages at this point are default installed ones, e.g. pip&setuptools
+            # And we want to upgrade them cleanly, i.e. uninstall them
+            'pip_ignore_installed': False,
+            # Python installations must be clean
+            'sanity_pip_check': True,
+        }
+        # Disable templating to update the underlying values
+        with disable_templating(self.cfg):
+            for key, value in ext_defaults.items():
+                if key not in self.cfg['exts_default_options']:
+                    self.cfg['exts_default_options'][key] = value
```

Comment:
You could simplify this a bit by using `get_ref`, you shouldn't need `disable_templating` anymore then:

```python
exts_default_options = self.cfg.get_ref('exts_default_options')
for key in ext_defaults:
    if key not in exts_default_options:
        exts_default_options[key] = ext_defaults[key]
```

Reply:
Good idea, done